### DISCUSSION TITLE
Translations for cms pages best practives

### DISCUSSION THREAD

**Discussion by km37500** [directus/directus#4432] [Archive]
Hi,

I have cms pages (about, services... ) collection, with fields like title, content, header-image, meta_title, meta_keywords, meta_description. The site is in 5 languages.

What is the best way to handle that?

1
Make every field with translations
Then the layout of the single item is not clearly arranged and very long. Every field has 4 urls for editing the language content

2.
old-fashioned solution: make language field and dupplicate every page with language_field = 'en' ...

What would you suggest ?

Regards
M.

**Comment by rijkvanzanten** [comment]
I would suggest going with (1): having a single `pages` collection with a single `translations` in which you manage all the translated fields on the page. Next to that, I'd vote up https://github.com/directus/directus/discussions/3011 and leave your thoughts on how to improve the translations experience 🙂

**Reply by km37500** [reply]
If I have understood it right, we need for every field (title, content, header-image, meta_title, meta_keywords, meta_description) in the table "single-page" a new table? or are they combined in one table?

Besides that: is there a translation file (CSV) possible for translations of input labels or labels of input values?

**Reply by benhaynes** [reply]
You would have ONE "translations" collection for each "normal" collection. So `articles` (containing all language agnostic fields) would have one `article_translations` collection with all the translated fields within it.
